analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp8.cs

```diff
@@ -82,3 +82,65 @@ private class MyServiceTransient : IMyService
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/6653
+namespace Repro6653
+{
+    public class UsingCoalescingAssignmentOperator
+    {
+        private static string StaticField1; // Compliant, read and written by ??= in SomeStaticMethod
+
+        private string Field0_1;            // Noncompliant, never read nor written
+        private string Field0_2 = "1";      // Noncompliant, written by initializer but not read
+        private string Field0_3 = "1";      // Compliant, read by Property0 and written by initializer
```

Comment:
I would remove all the `// Compliant` comments for maintainability reasons. This class is focused on coalescing and quick Ctrl+F shows the usage and reason pretty clearly.